packages/brain/src/modules/cron/sendProofsOfValidation.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  DappnodeSignatureVerifierPostRequest,
+  Web3signerPostSignDappnodeRequest,
+  Web3signerPostSignDappnodeResponse,
+} from "@stakingbrain/common";
+import {
+  Web3SignerApi,
+  DappnodeSignatureVerifier,
+} from "../apiClients/index.js";
+import { BrainDataBase } from "../db/index.js";
+import logger from "../logger/index.js";
+
+/**
+ * Send the proof of validation to the dappnode-signatures.io domain
+ */
+export async function sendProofsOfValidation(
+  signerApi: Web3SignerApi,
+  brainDb: BrainDataBase,
+  DappnodeSignatureVerifier: DappnodeSignatureVerifier,
+  shareDataWithDappnode: boolean
+): Promise<void> {
+  try {
+    // Get the proofs of validation from the signer
+    const proofsOfValidations = await getProofsOfValidation(
+      signerApi,
+      brainDb,
+      shareDataWithDappnode
+    );
+    if (proofsOfValidations.length === 0) {
+      logger.debug(`No proofs of validation to send`);
+      return;
+    }
+    logger.debug(`Sending ${proofsOfValidations.length} proofs of validations`);
+    await DappnodeSignatureVerifier.sendProofsOfValidation(proofsOfValidations);
+  } catch (e) {
+    logger.error(`Error sending proof of validation: ${e.message}`);
+  }
+}
+
+/**
+ * Get the proofs of validation from the signer
+ * for all the pubkeys in the db
+ */
+async function getProofsOfValidation(
+  signerApi: Web3SignerApi,
+  brainDb: BrainDataBase,
+  shareDataWithDappnode: boolean
+): Promise<DappnodeSignatureVerifierPostRequest[]> {
+  const signerDappnodeSignRequest: Web3signerPostSignDappnodeRequest = {
+    type: "PROOF_OF_VALIDATION",
+    platform: "dappnode",
+    timestamp: Date.now().toString(),
+  };
+  // get pubkeys detauls from db
+  const dbPubkeysDetails = brainDb.getData();
+
+  // only send proof of validation if the user has enabled it
+  // or if there is a stader pubkey
+  if (
+    !shareDataWithDappnode &&
+    !Object.values(dbPubkeysDetails).some((pubkey) => pubkey.tag === "stader")
```

Comment:
But here if there is a stader pubkey will the brain send the validation proofs for all the pubkeys even if shareDataWithDappnode is disabled?